soc/xlnx/versal/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2025 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+zephyr_sources_ifdef(
+  CONFIG_ARM_MMU
+  arm_mmu_regions.c
+)
+
+zephyr_include_directories(.)
+
+if(CONFIG_SOC_AMD_VERSAL_APU)
```

Comment:
This does not seem to be absolutely necessary - this directory is only included when enabling this SoC. Or is this port supposed to support other cores, like a secondary AArch32 core?

Reply:
This soc has RPU as well will add that support on top of this that's why kept this if check

Reply:
Thank you for explaining, this makes sense now